src/Cassandra/Compression/LZ4Compressor.cs

```diff
@@ -25,13 +25,39 @@ internal class LZ4Compressor : IFrameCompressor
         public Stream Decompress(Stream stream)
         {
             var buffer = Utils.ReadAllBytes(stream, 0);
+            if (buffer.Length < 4)
+            {
+                throw new CorruptionException("Corrupt literal length");
```

Comment:
I would change this exception message to be a bit more specific like `Unable to parse uncompressed size due to invalid buffer length`